### DISCUSSION TITLE
Rest API (PATCH) not working as expected

### DISCUSSION THREAD

**Discussion by MrSTOP** [seriousm4x/UpSnap#884] [Help wanted] [answered]
I want to get the current status of a computer using http request in node red. 
I have been able to get the computer status by using ```[GET] /api/collections/devices/records/:id```, but the status does not update. According to the Wiki, I need to use ```[PATCH] /api/collections/devices/records/:id```to update the computer status. When I send the request, the "updated" time in the http response will change, but the "status" in the response will not update. The computer status will only be updated when I open the upsnap webpage.
How should I update the computer status through the Rest API?

**Comment by seriousm4x** [comment]
> but the status does not update

Sounds like you have "Lazy ping" enabled. When lazy ping is enabled, the device status will only be updated when a client is connected to the webinterface.

> When I send the request, the "updated" time in the http response will change, but the "status" in the response will not update.

It's not really recommended to change the "status" value of a device since it well be overwritten as soon as upspan runs another ping.
Any specific reason why you want to change the status via the api?

**Reply by MrSTOP** [reply]
Thanks for your reply.
I don't want to change the device status, I want to update the device status, because I want to get the current status of the device through the API and display it, but the device status  obtained by ```[GET] /api/collections/devices/records/:id``` will only be updated when I power on/sleep through the web page.
So should I turn off lazy ping?

**Reply by MrSTOP** [reply, marked as answer]
Turning off lazy ping solved my problem
